### PR TITLE
Fix `dask-core` pinning to properly exclude nightlies

### PR DESCRIPTION
**Pull request by charlesbluca**
As brought up in https://github.com/conda-forge/dask-feedstock/pull/190#issuecomment-1215624680, the pinning introduced in #217 accidentally blocked the security backport proposal - tested this new pinning out locally and it seems to work with backports while also blocking nightly packages.

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

**Review by jrbourbeau — COMMENTED** [review]
Thanks @charlesbluca! Seems reasonable to me -- @jakirkham do you have a minute to look at this?

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca -- will plan to merge this at EOD if no further comments

**Comment by jakirkham** [comment]
Have the same feedback as in the other review ( https://github.com/conda-forge/dask-feedstock/pull/191#pullrequestreview-1077933301 ). Would suggest we keep this discussion in that thread for now and then apply whatever we decide to both PRs

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@jrbourbeau, could you please take a look at this as well?

If this seems reasonable, would like to merge it before we merge the release PR. That way the new release will have the right version constraint to start 🙂

**Pull request merged**
